asf_public_discourse_home_decarbonisation/utils/text_processing_utils.py

```diff
@@ -0,0 +1,205 @@
+"""
+Utils for processing text data.
+"""
+
+import re
+from nltk.corpus import stopwords
+from nltk import ngrams
+import nltk
+
+nltk.download("stopwords")
+from gensim.parsing.preprocessing import STOPWORDS
+import re
+import string
+from nltk.stem.wordnet import WordNetLemmatizer
+from nltk.tag import pos_tag
+
+from nltk import pos_tag
+from nltk.corpus import wordnet
+
+nltk.download("punkt")
+nltk.download("averaged_perceptron_tagger")
+nltk.download("wordnet")
+import pandas as pd
+from collections import Counter
+
+
+def remove_urls(text: str) -> str:
+    """
+    Removes URLs from text.
+    Args:
+        text: a string, tipically one or multiple sentences long
+    Returns:
+        text without URLs
+    """
+    pattern = r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+    cleaned_text = re.sub(pattern, " ", text)
+    return cleaned_text
+
+
+def replace_punctuation_with_space(text: str) -> str:
+    """
+    Replaces punctuation with space.
+    Args:
+        text: a string, tipically one or multiple sentences long
+    Returns:
+        text wihtout punctuation
+    """
+    pattern = r"[{}]".format(re.escape(string.punctuation))
+    replaced_text = re.sub(pattern, " ", text)
+    replaced_text = re.sub(r"\s+", " ", replaced_text)  # Remove consecutive spaces
+    return replaced_text.strip()  # Remove leading/trailing spaces
+
+
+def remove_text_after_patterns(text: str) -> str:
+    """
+    Removes pattern of the form "xxx writes: ".
+
+    Args:
+        text (str): text to be cleaned
+
+    Returns:
+        str: cleaned text
+    """
+    # We use re.sub() to replace the pattern with an empty string
+    result = re.sub(r"\w+ wrote »", " ", text)
+    return result
+
+
+def identify_part_of_spech(pos_tag: str) -> str:
+    """
+    Convert the tag given by nltk.pos_tag to the tag used by wordnet.
+
+    Args:
+        pos_tag (str): NLTk part of speach tag
+
+    Returns:
+        str: WordNet part os speach tag
+    """
+    if pos_tag.startswith("N"):
+        return wordnet.NOUN
+    elif pos_tag.startswith("V"):
+        return wordnet.VERB
+    elif pos_tag.startswith("R"):
```

Comment:
```suggestion
    elif pos_tag.startswith("RB"):
```
should this be RB? [Ref](https://www.ling.upenn.edu/courses/Fall_2003/ling001/penn_treebank_pos.html)

Reply:
I think you're right, otherwise we're also catching particles, not just adverbs.

Reply:
Fixed :)